### PR TITLE
mobile: adding a fake upstream for TCP tests

### PR DESCRIPTION
**Pull request by alyssawilk**
As an alternate to https://github.com/envoyproxy/envoy/pull/25552 allow creating a TCP test upstream which will respond to requests.

Risk Level: low
Testing: yes
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

**Comment by adisuissa** [comment]
Assigning Ryan for now, but feel free to reassign to another mobile maintainer.
/assign @RyanTheOptimist

File: mobile/test/common/integration/quic_test_server.h [hunk 1]
```diff
@@ -24,24 +25,29 @@ class QuicTestServer {
   Thread::SkipAsserts skip_asserts_;
   ProcessWide process_wide;
   Thread::MutexBasicLockable lock;
+  ;
```
Comment (RyanTheOptimist) [inline mobile/test/common/integration/quic_test_server.h:28]: nit: remove this?

**Review by RyanTheOptimist — APPROVED** [review]
Wow, nice! One tiny nit.

File: mobile/test/common/integration/quic_test_server.h [hunk 2]
```diff
@@ -10,8 +10,9 @@
 #include "test/mocks/server/transport_socket_factory_context.h"
 #include "test/config/utility.h"
 
+// TODO(alyssawilk) move this to a non-quic file name.
```
Comment (jpsim) [inline mobile/test/common/integration/quic_test_server.h:13]: Can we do this before merging please?

**Review by jpsim — APPROVED** [review]
+1 for this change as-is, it's useful to have real networking tests for Kotlin/Android.

However as discussed in Slack, I don't think this can serve as an alternative to https://github.com/envoyproxy/envoy/pull/25552 because then external consumers will need to pull in an additional dependency from Envoy Mobile in addition to the main library.

**Pull request merged**